extensions/namespace-lookup/src/main/java/io/druid/server/namespace/cache/NamespaceExtractionCacheManager.java

```diff
@@ -447,32 +451,54 @@ public void onFailure(Throwable t)
         }
         throw Throwables.propagate(e);
       }
-      Futures.addCallback(
-          future, new FutureCallback<Object>()
-          {
-            @Override
-            public void onSuccess(@Nullable Object result)
-            {
-              log.debug("Namespace [%s] successfully updated", namespaceName);
-            }
 
-            @Override
-            public void onFailure(Throwable t)
-            {
-              delete(cacheId);
-              if (t instanceof CancellationException) {
-                log.debug(t, "Namespace [%s] cancelled", namespaceName);
-              } else {
-                log.error(t, "Failed update namespace [%s]", namespace);
-              }
-            }
+      final CallbackWaiter<Object> lastCallback = new CallbackWaiter<Object>()
+      {
+        private boolean done;
+
+        @Override
+        public synchronized boolean isDone()
```

Comment:
does this need to be synchronized? Wouldn't it be enough for the variable to be volatile?